File: 卡码/代码随想录/C++/哈希/03-快乐数/LeetCode_0202.cpp
//202. ¿ìÀÖÊı
#include <iostream>
#include <unordered_set>
using namespace std;

class Solution {
public:
  bool isHappy(int n) {
		unordered_set<int> s;
		while(n!=1&&s.find(n)==s.end()){
			s.insert(n);
			n=getSum(n);
		}
		return n==1;
  }
  int getSum(int n){
  	int sum=0;
		while(n>0){
			sum+=(n%10)*(n%10);
			n/=10;
		}
		return sum;
	}
};

File: 卡码/代码随想录/Java/codeThoughts/src/main/java/com/kurumi/leetCode/codeThoughts/哈希/_03_快乐数/LeetCode_0202.java
package com.kurumi.leetCode.codeThoughts.哈希._03_快乐数;

import java.util.HashSet;

/**
 * @Author kurumi
 * @Date 2026-07-03 上午11:52:20
 * @Description 202. 快乐数
 */
public class LeetCode_0202 {
    static class Solution {
        public boolean isHappy(int n) {
            HashSet<Integer> set = new HashSet<Integer>();
            // 如果n不为1且set中不包含n
            while (n != 1 && !set.contains(n)) {
                set.add(n);
                n = getSum(n);
            }
            return n == 1;
        }

        public int getSum(int n) {
            int sum = 0;
            while (n > 0) {
                sum += (n % 10) * (n % 10);
                n /= 10;
            }
            return sum;
        }
    }
}
